(maint) Clarify that --config is optional to migrate

justinstoller opened the pull request.

justinstoller commented:
Do not merge. Pending feedback from community.

mwaggett commented on lib/puppetserver/ca/action/migrate.rb:
```diff
@@ -18,8 +18,9 @@ class Migrate
-  upgrading from Puppet Platform 6.x to Puppet 7. Use the currently configured
-  puppet.conf file in your installation, or supply one using the `--config` flag.
+  upgrading from Puppet Platform 6.x to Puppet 7. Uses the default puppet.conf
+  in your installation, override by supplying the `--config` flag.
```
maybe, if we wanna be super duper explicit, we could say:
```suggestion
  in your installation, or use a different config by supplying the `--config` flag.
```
?

mwaggett commented on lib/puppetserver/ca/cli.rb:
```diff
@@ -28,8 +28,8 @@ class Cli
-        'enable' => Action::Enable,
-        'migrate' => Action::Migrate,
+        'enable'   => Action::Enable,
+        'migrate'  => Action::Migrate,
```
I thought we decided we weren't going to do this kind of alignment? My IDE does it automatically, but I usually put it back, so I just wanna be clear on where we've landed with that.

justinstoller replied:
I don't think it's idiomatic to do it in clojure, I do think it's idiomatic in Ruby. I don't know that I'm the best at following that though.

mwaggett replied:
ahh, okay. I don't really care, just wasn't sure what we had agreed on.

mwaggett commented on lib/puppetserver/ca/cli.rb:
```diff
@@ -121,6 +121,9 @@ def self.parse_general_inputs(inputs)
           opts.on('--version', 'Display the version') do |v|
             parsed['version'] = true
           end
+          opts.on('--config', 'Path to a custom puppet.conf') do |v|
```
Was the `--config` option just totally ignored before?? Is there something else we need to do to ensure the custom config is used?

justinstoller replied:
......... weird... I did not mean to include this in my change.

mwaggett replied:
so.. are you saying this shouldn't be here? How does passing `--config` work now?

justinstoller replied:
Each action defines it.
At one point I thought that was dumb and started to do a maint to refactor it all into the cli class, but didn't finish it. Somehow when I checked out this branch some of that work came along?

mwaggett replied:
ahh, gotcha. so it sounds like this should be removed.

mwaggett commented:
Are you planning to update this @justinstoller ?

justinstoller commented:
Planning's a strong word, I at one point intended to.

I've removed the code that sneaked in besides the doc update and added your suggestion to the wording.

mwaggett commented:
not sure what's up with all those test failures..

Magisus commented:
That's still the Facter bug. It's been fixed, but this pulls in only released versions. I guess they haven't done a release yet.
